### ISSUE TITLE
Binder example can get stuck at uniform sequence due to symmetric logit initialization and unused Gumbel noise

### ISSUE THREAD

**Issue by henryhardigan** [rkruegs123/idp-design#4]
I ran into what looks like a symmetry issue in the binder example, the binder logits are initialized as:

init_logits = onp.full((binder_length, 20), 100.0)

and normalize() currently does:

def normalize(logits, temp, norm_key):
    gumbel_weights = jax.random.gumbel(norm_key, logits.shape)
    # pseq = jax.nn.softmax((logits + gumbel_weights) / temp)
    pseq = jax.nn.softmax(logits / temp)
    return pseq

A few things suggest that Gumbel-softmax was originally intended here:
norm_key is passed in only to generate gumbel_weights
gumbel_weights is computed but not used
the script explicitly defines and anneals gumbel_start, gumbel_end, and gumbel_temps

Because all binder logits start identically, and the current code uses deterministic softmax without the Gumbel perturbation, the initial binder pseq is exactly uniform. 

In my tests this can leave the optimization stuck in a fully symmetric state:
binder pseq remains uniform
argmax sequence is arbitrary / tie-broken
gradients on the logits are zero
the apparent changes in the objective come only from resampling, not from sequence learning

A possible fix would be to break symmetry in one or both of these ways:

Re-enable the existing Gumbel-softmax perturbation:
def normalize(logits, temp, norm_key):
    gumbel_weights = jax.random.gumbel(norm_key, logits.shape)
    pseq = jax.nn.softmax((logits + gumbel_weights) / temp)
    return pseq

Replace the constant logit initialization with a small random initialization, e.g.:
key, init_key = random.split(key)
init_logits = random.normal(init_key, (binder_length, 20), dtype=jnp.float64)
init_logits = init_logits * 0.01
params = {"logits": init_logits}

It may also be useful to expose the initialization scale as a CLI argument such as --init-logit-scale.

(At least that's what Codex thought the issue was, feel free to tell me if I'm way off base)

EDIT: The issue might've stemmed from some changes I introduced, I was attempting to make this compatible with flexible linkers.
